trace/trace.go

```diff
@@ -353,6 +356,22 @@ func (sc SpanContext) IsValid() bool {
 	return sc.HasTraceID() && sc.HasSpanID()
 }
 
+// IsRemote indicates whether the SpanContext represents a remotely-created Span.
+func (sc SpanContext) IsRemote() bool {
+	return sc.remote
+}
+
+// WithRemote returns a new SpanContext with the remote property replaced.
+func (sc SpanContext) WithRemote(remote bool) SpanContext {
```

Comment:
I'm wondering if we want to reuse the term "with" here in addition to our configuration standards (e.g. `WithAttributes`, `WithResource`) or if we want to introduce a new term for a method that returns a copy. The first thing that comes to mind is `AsRemote`.

Reply:
I'm not sure that would scan very well in context, particularly for attributes that aren't boolean:

```go
sc = sc.WithSpanID(spanID)
sc = sc.WithRemote(true)
```

vs.

```go
sc = sc.AsSpanID(spanID)
sc = sc.AsRemote(true)
```